### ISSUE TITLE
Linter should warn if manifest.json refers to a file that does not exist

### ISSUE THREAD

**Issue by cpeterso** [mozilla/addons-linter#1658]
### Describe the problem and steps to reproduce it:
I accidentally published an extension with a manifest.json whose `"icons": { "48": "icon.png"}` pointed to a file that didn't existed (because I had renamed the icon file).

### What happened?
addons-linter and web-ext silently accepted the manifest.json without warning.

### What did you expect to happen?
I expected a warning that the referenced files do not exist.

### Anything else we should know?
This problem doesn't just affected "icons" files in the manifest. I can reproduce it with `"background":{"scripts":"background-FILE-DOES-NOT-EXIST.js"}`.

**Comment by andymckay** [comment]
Hmm, I thought I fixed this for background pages in #1562.

I changed a test add-on and broke the icons and background scripts and got this output. Can I check what version you are on please?

```
every-event web-ext --version                                                                                                                     master  ✭ ✱ 
2.2.2
every-event web-ext lint                                                                                                                          master  ✭ ✱ 
Validation Summary:

errors          2              
notices         0              
warnings        0              

ERRORS:

Code                                 Message                           Description                                                 File            Line   Column
MANIFEST_ICON_NOT_FOUND              An icon defined in the manifest   Icon could not be found at "alarm.svga".                    manifest.json                
                                     could not be found in the                                                                                                  
                                     package.                                                                                                                   
MANIFEST_BACKGROUND_FILE_NOT_FOUND   A background script defined in    Background script could not be found at "background.jsa".   manifest.json                
                                     the manifest could not be                                                                                                  
                                     found.  
```

**Comment by cpeterso** [comment]
I'm using web-ext version 2.2.2 on macOS.

**Comment by andymckay** [comment]
Hmm, do you have your addon somewhere?

**Comment by cpeterso** [comment]
> Hmm, do you have your addon somewhere?

Yes. You can git clone https://github.com/cpeterso/clear-flash-cookies and run ./build.sh.

I've narrowed the problem somewhat: I see expected FILE_NOT_FOUND warnings when running `web-ext lint -s src` but not `addons-linter src`. Is addons-linter deprecated in favor of web-ext lint?

**Comment by andymckay** [comment]
addons-linter is a library, we use this as the source for web-ext and the linter on AMO. AMO gets updated every week, web-ext not quite so much. Currently addons-linter is on 0.28.2 and web-ext is on 0.27.0.

The next scheduled release of web-ext will be in the next week, we are doing a regularly scheduled export of the schema from Firefox when it merges to beta into the linter and then doing a new release.

So I expect this to be fixed in the next release of web-ext coming in about one week. It might be worth opening a bug on web-ext to make it clear what version of the linter it's using, something like `web-ext lint --version`.

Based on that I'm closing this and #1657.

**Comment by cpeterso** [comment]
Thanks. Just to be clear: `web-ext lint` version 2.2.2 is doing everything correctly! It shows the expected FILE_NOT_FOUND warnings I wanted to see and doesn't show the incorrect "Unknown permission: browsingData" warning (#1657).

The addons-linter command-line utility is what broken: it omits the expected FILE_NOT_FOUND warnings but shows the incorrect "Unknown permission: browsingData" warning. I will just use web-ext instead of addons-linter from now on.

**Comment by ValentinaPC** [comment]
Verified this on AMO-dev FF57(Win 7).
If a file is mentioned in the manifest, and does not exist, and error message is returned.
Postfix screenshot:
![2017-11-17_1339](https://user-images.githubusercontent.com/15685960/32945624-d03f166a-cb9c-11e7-9d1c-5c5fbb4c9865.png)
